dozer-sql/src/pipeline/expression/comparison.rs

```diff
@@ -1089,9 +1622,14 @@ pub fn evaluate_gt(
                 ">".to_string(),
             )),
         },
-        Field::Text(ref left_v) => match right_p {
-            Field::String(ref right_v) => Ok(Field::Boolean(left_v > right_v)),
-            Field::Text(ref right_v) => Ok(Field::Boolean(left_v > right_v)),
+        Field::Date(left_v) => match right_p {
+            Field::Date(right_v) => Ok(Field::Boolean(left_v > right_v)),
+            Field::String(right_v) | Field::Text(right_v) => {
+                let right_val = NaiveDate::from_str(right_v.as_str()).map_err(|_| {
+                    PipelineError::UnableToCast(right_v.to_string(), "Date".to_string())
+                })?;
```

Comment:
I think casting this automatically is a bit dangerous, we should keep the date format in consideration.
Probably an explicit cast is better in this case.

Reply:
fixed